src/seedu/addressbook/data/person/Name.java

```diff
@@ -60,5 +60,16 @@ public boolean equals(Object other) {
     public int hashCode() {
         return fullName.hashCode();
     }
-
+    
+    /**
+     * Returns true of the other name is very similar to this name.
+     * Two names are considered similar if:
+     * <ul>
+     * <li>One name is a case-insensitive subset/superset of the other and vice versa.</li>
+     * <li>Both names are non-null.</li>
+     * </ul>
+     */
+    public boolean isSimilar(Name other) {
```

Comment:
in TDD, you should create your test case first. If Java complains that the method is missing or having some syntax error, it is okay, because it is still considered failing test case. Take a look at the 3 Rules of TDD from Uncle Bob (link on repo home page).